stricmp is being deprecated in Cygwin.
One should use strcasecmp.
See https://cygwin.com/ml/cygwin/2014-10/msg00359.html
diff -druN a/latest/latest/Singular/run.c b/latest/latest/Singular/run.c
--- a/latest/Singular/run.c	2014-11-19 14:06:05.000000000 +0100
+++ b/latest/Singular/run.c	2015-01-16 09:32:45.771298300 +0100
@@ -45,6 +45,7 @@
  #include <sys/stat.h>
  #include <sys/cygwin.h>
  #include <sys/unistd.h>
+ #define stricmp strcasecmp
 //WinMainCRTStartup() { mainCRTStartup(); }
 #else
  #include <direct.h>
